recurly/base_client.py

```diff
@@ -76,3 +76,19 @@ def _interpolate_path(self, path, *args):
         self._validate_path_parameters(args)
 
         return path % tuple(map(lambda arg: urllib.parse.quote(arg, safe=""), args))
+
+    def _url_encode(self, params):
```

Comment:
I like this a lot better than the `__map_array_params` solution. We should consider doing the same in all of the clients.